classes/Cart.php

```diff
@@ -332,6 +332,32 @@ public function updateAddressId($id_address, $id_address_new)
         Db::getInstance()->execute($sql);
     }
 
+    /**
+     * Update the Delivery Address ID of the Cart.
+     *
+     * @param int $currentAddressId Current Address ID to change
+     * @param int $newAddressId New Address ID
+     */
+    public function updateDeliveryAddressId(int $currentAddressId, int $newAddressId)
+    {
+        if (!isset($this->id_address_delivery) || (int) $this->id_address_delivery === $currentAddressId) {
+            $this->id_address_delivery = $newAddressId;
+            $this->update();
+        }
+
+        $sql = 'UPDATE `' . _DB_PREFIX_ . 'cart_product`
+        SET `id_address_delivery` = ' . $newAddressId . '
+        WHERE  `id_cart` = ' . (int) $this->id . '
+            AND `id_address_delivery` = ' . $currentAddressId;
+        Db::getInstance()->execute($sql);
```

Comment:
Do we need a variable ? I would use sprintf here :D

Reply:
I just copied the initial function `Cart:updateAddressId` I don't particulary try to improve legacy code ^^